manifests/init.pp

```diff
@@ -120,7 +120,7 @@
     ip           => $localhost6_ip,
   }
 
-  if $use_fqdn_real == true {
+  if $::settings::storeconfigs and $use_fqdn_real == true {
```

Comment:
If use_fqdn is true and we are not using storeconfigs, there should still be the `host { $::fqdn` entry, though without the `@@`